templates/taxi/car_detail.html

```diff
@@ -2,21 +2,43 @@
 
 {% block content %}
   <h1>
-    {{ car.model }}
-    <a href="{% url 'taxi:car-delete' pk=car.id %}" class="btn btn-danger link-to-page">
-      Delete
-    </a>
-
-    <a href="{% url 'taxi:car-update' pk=car.id %}" class="btn btn-secondary link-to-page">
-      Update
-    </a>
+    Model: {{ car.model }}
   </h1>
   <p>Manufacturer: ({{ car.manufacturer.name }}, {{ car.manufacturer.country }})</p>
-  <h1>Drivers</h1>
+
+  <div class="row">
+    <div class="col-5">
+      <h3>Drivers</h3>
+    </div>
+    <div class="col-7">
+      <form action="{% url 'taxi:assign-driver' car.pk %}" method="post">
+        {% csrf_token %}
+        {% if request.user in car.drivers.all %}
+          <button type="submit" class="btn btn-secondary link-to-page">
+            Delete me from this car
+          </button>
+        {% else %}
+          <button type="submit" class="btn btn-success link-to-page">
+            Assign me to this car
+          </button>
+        {% endif %}
+      </form>
+    </div>
+  </div>
+
   <hr>
   <ul>
     {% for driver in car.drivers.all %}
       <li>{{ driver.username }} ({{ driver.first_name }} {{ driver.last_name }})</li>
     {% endfor %}
   </ul>
+
+  <a href="{% url 'taxi:car-update' pk=car.id %}"
+     class="btn btn-secondary">
+    Update
+  </a>
+  <a href="{% url 'taxi:car-delete' pk=car.id %}" class="btn btn-danger">
```

Comment:
Ensure that the URL names 'taxi:car-update' and 'taxi:car-delete' used in the links match the URL configuration in your Django project. If the URL patterns are named differently, update these to prevent navigation errors.